pkg/devfile/adapters/docker/component/utils_test.go

```diff
@@ -235,3 +238,113 @@ func TestStartContainer(t *testing.T) {
 	}
 
 }
+
+func TestGenerateAndGetHostConfig(t *testing.T) {
+	fakeClient := lclient.FakeNew()
+	testComponentName := "test"
+	componentType := versionsCommon.DevfileComponentTypeDockerimage
+
+	tests := []struct {
+		name         string
+		urlValue     []envinfo.EnvInfoURL
+		expectResult nat.PortMap
+		client       *lclient.Client
+	}{
+		{
+			name:         "Case 1: no port mappings",
+			urlValue:     []envinfo.EnvInfoURL{},
+			expectResult: nil,
+			client:       fakeClient,
+		},
+		{
+			name: "Case 2: only one port mapping",
+			urlValue: []envinfo.EnvInfoURL{
+				{Port: 8080, ExposedPort: 65432},
+			},
+			expectResult: nat.PortMap{
+				"8080/tcp": []nat.PortBinding{
+					{
+						HostIP:   LocalhostIP,
+						HostPort: "65432",
+					},
+				},
+			},
+			client: fakeClient,
+		},
+		{
+			name: "Case 3: multiple port mappings",
+			urlValue: []envinfo.EnvInfoURL{
+				{Port: 8080, ExposedPort: 65432},
+				{Port: 9090, ExposedPort: 54321},
+				{Port: 9080, ExposedPort: 45678},
+			},
+			expectResult: nat.PortMap{
+				"8080/tcp": []nat.PortBinding{
+					{
+						HostIP:   LocalhostIP,
+						HostPort: "65432",
+					},
+				},
+				"9090/tcp": []nat.PortBinding{
+					{
+						HostIP:   LocalhostIP,
+						HostPort: "54321",
+					},
+				},
+				"9080/tcp": []nat.PortBinding{
+					{
+						HostIP:   LocalhostIP,
+						HostPort: "45678",
+					},
+				},
+			},
+			client: fakeClient,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			devObj := devfileParser.DevfileObj{
+				Data: testingutil.TestDevfileData{
+					ComponentType: componentType,
+				},
+			}
+
+			adapterCtx := adaptersCommon.AdapterContext{
+				ComponentName: testComponentName,
+				Devfile:       devObj,
+			}
+
+			esi, err := envinfo.NewEnvSpecificInfo("")
+			if err != nil {
+				t.Error(err)
+			}
+			for _, element := range tt.urlValue {
+				err = esi.SetConfiguration("URL", element)
+				if err != nil {
+					t.Error(err)
+				}
```

Comment:
This will be a write operation. Instead you can just do 

```
settings:=esi.GetComponentSettings()
settings.URL=&tt.urlValue
```

Reply:
It has to write to the file. Since `getPortMap()` will read from the env.yaml file.

Reply:
OK :+1: